src/config_server.c

```diff
@@ -0,0 +1,2396 @@
+{
+    free(auth_client->pam_config_name);
+    auth_client->pam_config_name = NULL;
+}
+
+static void
+nc_server_del_auth_client_pam_dir(struct nc_client_auth *auth_client)
+{
+    free(auth_client->pam_config_dir);
+    auth_client->pam_config_dir = NULL;
+}
+
+static void
+nc_server_del_endpt_name(struct nc_endpt *endpt)
+{
+    free(endpt->name);
+    endpt->name = NULL;
+}
+
+static void
+nc_server_del_local_address(struct nc_bind *bind)
+{
+    free(bind->address);
+    bind->address = NULL;
+}
+
+static void
+nc_server_del_hostkey_name(struct nc_hostkey *hostkey)
+{
+    free(hostkey->name);
+    hostkey->name = NULL;
+}
+
+static void
+nc_server_del_public_key(struct nc_hostkey *hostkey)
+{
+    free(hostkey->pub_base64);
+    hostkey->pub_base64 = NULL;
+}
+
+static void
+nc_server_del_truststore_reference(struct nc_client_auth *client_auth)
+{
+    free(client_auth->ts_reference);
+    client_auth->ts_reference = NULL;
+}
+
+static void
+nc_server_del_private_key(struct nc_hostkey *hostkey)
+{
+    free(hostkey->priv_base64);
+    hostkey->priv_base64 = NULL;
+}
+
+static void
+nc_server_del_keystore_reference(struct nc_hostkey *hostkey)
+{
+    hostkey->keystore = NULL;
+}
+
+static void
+nc_server_del_auth_client_username(struct nc_client_auth *auth_client)
+{
+    free(auth_client->username);
+    auth_client->username = NULL;
+}
+
+static void
+nc_server_del_auth_client_pubkey_name(struct nc_client_auth_pubkey *pubkey)
+{
+    free(pubkey->name);
+    pubkey->name = NULL;
+}
+
+static void
+nc_server_del_auth_client_pubkey_pub_base64(struct nc_client_auth_pubkey *pubkey)
+{
+    free(pubkey->pub_base64);
+    pubkey->pub_base64 = NULL;
+}
+
+static void
+nc_server_del_auth_client_ts_reference(struct nc_client_auth *auth_client)
+{
+    free(auth_client->ts_reference);
+    auth_client->ts_reference = NULL;
+}
+
+static void
+nc_server_del_auth_client_password(struct nc_client_auth *auth_client)
+{
+    free(auth_client->password);
+    auth_client->password = NULL;
+}
+
+static void
+nc_server_del_hostkey_algs(struct nc_server_ssh_opts *opts)
+{
+    free(opts->hostkey_algs);
+    opts->hostkey_algs = NULL;
+}
+
+static void
+nc_server_del_kex_algs(struct nc_server_ssh_opts *opts)
+{
+    free(opts->kex_algs);
+    opts->kex_algs = NULL;
+}
+
+static void
+nc_server_del_encryption_algs(struct nc_server_ssh_opts *opts)
+{
+    free(opts->encryption_algs);
+    opts->encryption_algs = NULL;
+}
+
+static void
+nc_server_del_mac_algs(struct nc_server_ssh_opts *opts)
+{
+    free(opts->mac_algs);
+    opts->mac_algs = NULL;
+}
+
+static void
+nc_server_del_hostkey(struct nc_server_ssh_opts *opts, struct nc_hostkey *hostkey)
+{
+    if (hostkey->ks_type == LOCAL_DEFINITON) {
+        nc_server_del_public_key(hostkey);
+        nc_server_del_private_key(hostkey);
+    } else if (hostkey->ks_type == KEYSTORE_REFERENCE) {
+        nc_server_del_keystore_reference(hostkey);
+    } else {
+        return;
```

Comment:
Can there be another value? Is it just a safety for future changes? Depending on the answers, change this to failed asserts or error or something better, this code is not suitable.